packages/blocks/src/__internal__/utils/common-operations.ts

```diff
@@ -1,32 +1,30 @@
-import { matchFlavours } from '@blocksuite/global/utils';
+import { assertExists, matchFlavours } from '@blocksuite/global/utils';
 import type { Page } from '@blocksuite/store';
 import type { BaseBlockModel } from '@blocksuite/store';
 import type { VEditor, VRange } from '@blocksuite/virgo';
 
-import { getRichTextByModel } from './query.js';
+import { asyncGetRichTextByModel, getRichTextByModel } from './query.js';
 import type { ExtendedModel } from './types.js';
 
+export function asyncSetVRange(model: BaseBlockModel, vRange: VRange) {
+  asyncGetRichTextByModel(model)
+    .then(richText => {
```

Comment:
Can `async/await` syntax be used here, I think the code format will be more uniform

Reply:
I didn't use the asyn function here is because I want to handle the promise error in the function and make sure when it is called, users won't need to handle it manually.

Reply:
I think it is equivalent to write like this:

```ts
async function asyncSetVRange(model: BaseBlockModel, vRange: VRange) {
  try {
    const richText = await asyncGetRichTextByModel(model)
    richText?.vEditor?.setVRange(vRange);
  } catch (e) {
    throw e;
  }
}
```

Reply:
This try/catch looks redundant.

Reply:
> I think it is equivalent to write like this:
> 
> 
> 
> ```ts
> 
> async function asyncSetVRange(model: BaseBlockModel, vRange: VRange) {
> 
>   try {
> 
>     const richText = await asyncGetRichTextByModel(model)
> 
>     richText?.vEditor?.setVRange(vRange);
> 
>   } catch (e) {
> 
>     throw e;
> 
>   }
> 
> }
> 
> ```

IMO they won't be equal. When you're using try catch in an async function, it will reject the error, and if the place where it called doesn't catch it, it will be a floating promise error. But, nevermind, it's not that important in this case.